xbmc/cores/VideoPlayer/VideoRenderers/windows/RendererDXVA.cpp

```diff
@@ -32,8 +32,20 @@ void CRendererDXVA::GetWeight(std::map<RenderMethod, int>& weights, const VideoP
   unsigned weight = 0;
   const AVPixelFormat av_pixel_format = picture.videoBuffer->GetFormat();
 
+  const bool streamIsHDR = (picture.color_primaries == AVCOL_PRI_BT2020) &&
+                           (picture.color_transfer == AVCOL_TRC_SMPTE2084 ||
+                            picture.color_transfer == AVCOL_TRC_ARIB_STD_B67);
+  const auto settings = CServiceBroker::GetSettingsComponent()->GetSettings();
+  const auto setting = DX::Windowing()->SETTING_WINSYSTEM_IS_HDR_DISPLAY;
+  const bool systemUsesHDR =
+      settings ? settings->GetBool(setting) && DX::Windowing()->IsHDRDisplay() : false;
```

Comment:
HDR logic in Windows is quite complicated and this not covers all use cases: setting in Kodi can be off but HDR enabled permanently in Windows display settings or user can toggle HDR on manually with F11 key before start movie...

Then is need:
` (settings->GetBool(setting) && DX::Windowing()->IsHDRDisplay()) || DX::Windowing()->IsHDROutput()`